task/sql-tasks.js

```diff
@@ -280,7 +432,23 @@ async function task_1_21(db) {
  * @return {array}
  */
 async function task_1_22(db) {
-    throw new Error("Not implemented");
+    let result = await db.query(`
+        SELECT DISTINCT
+            Customers.CompanyName,
+            Products.ProductName,
+            OrderDetails.UnitPrice AS PricePerItem
+        FROM Customers, Orders, OrderDetails, Products
+        WHERE Customers.CustomerId = Orders.CustomerId AND 
+            Orders.OrderId = OrderDetails.OrderId AND 
+            OrderDetails.ProductId = Products.ProductId AND
+            OrderDetails.UnitPrice = (SELECT MAX(OrderDetails.UnitPrice)

```

Comment:
You can move the complex Where condition as part of SELECT and avoid DISTINCT function also